Source/Events.Processing/Streams/AbstractScopedStreamProcessor.cs

```diff
@@ -116,6 +119,14 @@ public Task Start(CancellationToken cancellationToken)
         /// <returns>A <see cref="Task" /> that, when resolved, returns the new <see cref="IStreamProcessorState" />.</returns>
         protected abstract Task<IStreamProcessorState> OnFailedProcessingResult(FailedProcessing failedProcessing, StreamEvent processedEvent, IStreamProcessorState currentState);
 
+        /// <summary>
+        /// Tries to get the <see cref="TimeSpan" /> for when to retry processing. Outputs the maximum value <see cref="TimeSpan" /> if there is no retry time.
+        /// </summary>
+        /// <param name="state">The current <see cref="IStreamProcessorState" />..</param>
+        /// <param name="timeToRetry">The <see cref="TimeSpan" /> for when to retry processsing a stream processor.</param>
+        /// <returns>A value indicating whether there is a retry time.</returns>
+        protected abstract bool TryGetTimeToRetry(IStreamProcessorState state, out TimeSpan timeToRetry);
```

Comment:
```suggestion
        /// <summary>
        /// Tries to get the <see cref="TimeSpan" /> for when to retry processing. Outputs the maximum value <see cref="TimeSpan" /> if there is no retry time.
        /// </summary>
        /// <param name="state">The current <see cref="IStreamProcessorState" />..</param>
        /// <param name="timeToRetry">The <see cref="TimeSpan" /> for when to retry processsing a stream processor.</param>
        /// <returns>A value indicating whether there is a retry time.</returns>
        protected bool TryGetTimeToRetry(IStreamProcessorState state, out TimeSpan timeToRetry)
            =>  _timeToRetryGetter.TryGetTimespanToRetry(state as StreamProcessorState, out timeToRetry);
```

As the code for both of the implementing streamprocessors is the same here, we could just pass the `ICanGetTimeToRetryFor<T>` into the `AbstractScopedStreamProcessor`  and have it call the interface.

This would also require changes to the stream processors constructors.

Reply:
Cannot do that as that would make the AbstractScopedStreamProcessor generic, which I don't want to do since it would bubble down in lots of other parts of the code

Reply:
ok wont fix